include/pinocchio/bindings/python/spatial/inertia.hpp

```diff
@@ -295,6 +312,251 @@ namespace pinocchio
+#endif
+        bp::class_<PseudoInertia, HolderType>(
+          "PseudoInertia",
+          "This class represents a pseudo inertia matrix and it is defined by its mass, vector "
+          "part, and 3x3 matrix part.\n\n"
+          "Supported operations ...",
+          bp::no_init)
+          .def(PseudoInertiaPythonVisitor<PseudoInertia>())
+          .def(CopyableVisitor<PseudoInertia>())
+          .def(PrintableVisitor<PseudoInertia>());
+      }
+
+    private:
+      static Matrix4 __array__(const PseudoInertia & self, bp::object)
+      {
+        return self.toMatrix();
+      }
+
+      struct Pickle : bp::pickle_suite
+      {
+        static boost::python::tuple getinitargs(const PseudoInertia & pi)
+        {
+          return bp::make_tuple(pi.mass, pi.h, pi.sigma);
+        }
+
+        static bool getstate_manages_dict()
+        {
+          return true;
+        }
+      };
+
+    }; // struct PseudoInertiaPythonVisitor
+
+    template<typename LogCholeskyParameters>
+    struct LogCholeskyParametersPythonVisitor
+    : public boost::python::def_visitor<LogCholeskyParametersPythonVisitor<LogCholeskyParameters>>
+    {
+      enum
+      {
+        Options = LogCholeskyParameters::Options
+      };
+      typedef typename LogCholeskyParameters::Scalar Scalar;
+      typedef typename LogCholeskyParameters::Vector10 Vector10;
+      typedef typename LogCholeskyParameters::Matrix10 Matrix10;
+      typedef Eigen::Matrix<Scalar, Eigen::Dynamic, 1, Options> VectorXs;
+      typedef Eigen::Matrix<Scalar, Eigen::Dynamic, Eigen::Dynamic, Options> MatrixXs;
+
+    public:
+      template<class PyClass>
+      void visit(PyClass & cl) const
+      {
+        cl.def(bp::init<const VectorXs &>((bp::arg("self"), bp::arg("log_cholesky_parameters"))))
+          .def(bp::init<const LogCholeskyParameters &>(
+            (bp::arg("self"), bp::arg("clone")), "Copy constructor"))
+
+          .add_property(
+            "parameters", &LogCholeskyParametersPythonVisitor::getParameters,
+            &LogCholeskyParametersPythonVisitor::setParameters, "Log Cholesky parameters.")
+
+          .def(
+            "toDynamicParameters", &LogCholeskyParametersPythonVisitor::toDynamicParameters_proxy,
+            bp::arg("self"), "Returns the dynamic parameters representation.")
+          .def(
+            "toPseudoInertia", &LogCholeskyParameters::toPseudoInertia, bp::arg("self"),
+            "Returns the Pseudo Inertia representation.")
+          .def(
+            "toInertia", &LogCholeskyParameters::toInertia, bp::arg("self"),
+            "Returns the Inertia representation.")
+          .def(
+            "calculateJacobian", &LogCholeskyParametersPythonVisitor::calculateJacobian_proxy,
+            bp::arg("self"), "Calculates the Jacobian of the log Cholesky parameters.")
+
+          .def("__array__", &LogCholeskyParameters::toDynamicParameters)
+          .def("__array__", &__array__)
```

Comment:
Array definition have changed to support numpy 2 https://github.com/stack-of-tasks/pinocchio/blob/devel/include/pinocchio/bindings/python/spatial/se3.hpp#L200-L202